### PR TITLE
Migration guide for https://github.com/flutter/flutter/pull/47177

### PR DESCRIPTION
**Pull request by dnfield**
https://github.com/flutter/flutter/pull/47177 is a breaking change no matter how I try to slice it - even if I pull out the test related changes.  Provides a guide for migrating tests that this breaks.

### CODE REVIEW

**Comment by dnfield** [comment]
CI appears to be failing on unrelated files...

File: src/docs/release/breaking-changes/test-text-input.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The Flutter engine and framework have to keep state about text fields in sync.
+Sometimes, the engine can lose this state - such as when the application is
+backgrounded on Android. In this case, the engine asks the framework to re-send
+text field related state. [Currently, the framework is failing to send the
+actual text content of the text field in this scenario](https://github.com/flutter/flutter/issues/47137).
+
+## Description of change
+
+The Flutter framework tracks when the editing state of an `EditableText` widget
+changes. It keeps synchronizes this value to the engine, but only if the value
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/test-text-input.md]: Delete "keeps"

File: src/docs/release/breaking-changes/test-text-input.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+has changed. The state of this value is tracked by the `TextInputClient`, and
+logic exists to prevent sending duplicative values to the engine side.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/test-text-input.md]: duplicative => duplicate

File: src/docs/release/breaking-changes/test-text-input.md (hunk 1 above)
Comment (dnfield) [reply src/docs/release/breaking-changes/test-text-input.md]: Done

File: src/docs/release/breaking-changes/test-text-input.md (hunk 2 above)
Comment (dnfield) [reply src/docs/release/breaking-changes/test-text-input.md]: Done

**Comment by dnfield** [comment]
@Sfshaza - this is against the website repo :)  I've linked to the related flutter/flutter PR here.

**Comment by dnfield** [comment]
Thanks - let's give Hixie and goderbauer a chance to see it :)

File: src/docs/release/breaking-changes/test-text-input.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Tests that inspect the `TestTextInput.editingState` may have been able to pass
+before this change because:
+
+- There was dirty state from a previous test.
+- The framework did not call `setEditingState` on the `SystemChannels.textInput`
+  channel where it now does.
+
+## Migration guide
+
+Tests that relied on the dirty state will have to be updated. For example, this
```
Comment (goderbauer) [inline src/docs/release/breaking-changes/test-text-input.md]: maybe call out ".. dirty state from a previously run test..."?

Comment (dnfield) [reply src/docs/release/breaking-changes/test-text-input.md]: Done

**Pull request merged**